tools/ci-scripts/windows/Install-Stack.ps1

```diff
@@ -0,0 +1,31 @@
+[CmdletBinding()]
+param ( )
+
+Set-StrictMode -Version Latest
+$ErrorActionPreference = 'Stop'
+
+Add-Type -AssemblyName 'System.IO.Compression.FileSystem'
+
+$OutputDirectory = [System.IO.Path]::Combine(
+    [System.IO.Path]::GetTempPath(),
+    [System.IO.Path]::GetRandomFileName())
+
+if (-not (Test-Path -LiteralPath $OutputDirectory -PathType Container))
+{
+    mkdir $OutputDirectory | Out-Null
+}
+
+$stackInstaller = [System.IO.Path]::Combine($OutputDirectory, 'stack-x86_64.zip')
+Write-Debug "Downloading to $stackInstaller"
+
+[System.Net.ServicePointManager]::SecurityProtocol='TLS12'
+Invoke-WebRequest `
+    -Uri https://www.stackage.org/stack/windows-x86_64 `
+    -OutFile $stackInstaller
+
+Write-Debug "Extracting to $OutputDirectory"
```

Comment:
The script should download and extract in different locations so that we don't end up with the installer ZIP in the same directory as the final tool. Something like

```
$InstallDirectory = [System.IO.Path]::Combine($OutputDirectory, 'inst')
```